src/Hadolint/Rules.hs

```diff
@@ -335,6 +337,41 @@ hasNoMaintainer = instructionRule code severity message check
     check (Maintainer _) = False
     check _ = True
 
+hasMaintainerLabel :: Rule
+hasMaintainerLabel dockerfile = instructionRuleState code severity message check [] dockerfile
```

Comment:
Can you please make the state a `Set Linenumber Text` instead of a plain list. Lists are not great for performance

Reply:
Agree. 
I was looking for a consistent way of approaching a state of such scenarios. I see the map being used for another PR for health checks. 
So I have used the map here as well so that we can have a consistent approach for maintaining such state.